use-cases.md

```diff
@@ -53,11 +53,7 @@ When a server wants to trace execution of a request, it generally needs to go th
 Let's assume that we have an HTTP server, and the Span is propagated from the client via HTTP headers, accessible via `request.headers`:
 
 ```python
-    span = tracer.join_trace_from_text(
-        operation_name=operation,
-        context_snapshot=request.headers,
-        trace_attributes=request.headers
-    )
+    span = tracer.get_span_extractor(request.class).extract(request)
```

Comment:
> I thought your request.class was something like HttpRequest.class

That is correct and is what I meant in the initial examples, but it's not restricted to specific rpc request types. The example of using map.class or byte_buffer.class was raised though to illustrate this.

An example of rpc request based injector/extractors being thin is…

    public class HttpRequestSpanInjector implements SpanInjector {
         public void inject(Span span, HttpServletRequest request) {
              tracer.getSpanInjector(Map.class).inject(span, request.getheaders());
        }
    }

> The main concern people will have with that is the same one as on the setTag(String, Object) signature. 

Yes. Throwing a NoInjectorFoundException would be normal part of the Specfication.

     // throws a NoInjectorFoundException
     tracer.getSpanInjector(SomethingCompletelyWrong.class)

> So flexibility in this regard hurts portability.

get_span_injector(map.class) and get_span_injector(byte_buffer.class) is intended to solve that problem (and application code can plug in its own injectors and extractors)